### ISSUE TITLE
Check for underscore in variables

### ISSUE THREAD

**Issue by markbrummel** [markbrummel/vscode-allint#19]
Underscore should not be used in AL variables. Neither should spaces and other special characters.

**Comment by StefanHovenbitzer** [comment]
So basically only [A-Za-z0-9] are allowed? Wouldnt it be easier to just check for what is allowed in order to deny everything else? As i can see from the latest commit from mrabus the german letters "äüöß" are still allowed.

**Comment by markbrummel** [comment]
True. Good point. I actually wanted to check if the variable name matches the naming standards of Microsoft and include approved abbreviations. Let me put that on the issue list.

Great feedback

**Comment by mrabus** [comment]
Hi Mark, 

did you talk about these guidelines?
Source: [https://blogs.msdn.microsoft.com/nav/2015/01/09/cal-coding-guidelines-used-at-microsoft-development-center-copenhagen/](url)

Variable naming
Variables that refer to a C/AL object must contain the objects name, abbreviated where necessary.
A variable must begin with a capital letter.
Blanks, periods, and other characters (such as parentheses) that would make quotation marks
around a variable necessary must be omitted.
If a variable is a compound of two or more words or abbreviations, each word or abbreviation
should begin with a capital letter.

Temporary variable naming
The name of a temporary variable must be prefixed with the word Temp and not otherwise

TextConst variable names should have a suffix (an approved three-letter suffix: Msg, Tok, Err, Qst,
Lbl, Txt) describing usage.

I guess there is some work to be done :)

**Comment by markbrummel** [comment]
Yes. Let’s see how we can implement those while letting people change the rules if they want to.

 

Van: Matthias Rabus [mailto:notifications@github.com] 
Verzonden: woensdag 11 oktober 2017 12:31
Aan: markbrummel/vscode-allint <vscode-allint@noreply.github.com>
CC: Mark Brummel <brummelds@gmail.com>; Author <author@noreply.github.com>
Onderwerp: Re: [markbrummel/vscode-allint] Check for underscore in variables (#19)

 

Hi Mark,

did you talk about these guidelines?
Source: https://blogs.msdn.microsoft.com/nav/2015/01/09/cal-coding-guidelines-used-at-microsoft-development-center-copenhagen/

Variable naming
Variables that refer to a C/AL object must contain the objects name, abbreviated where necessary.
A variable must begin with a capital letter.
Blanks, periods, and other characters (such as parentheses) that would make quotation marks
around a variable necessary must be omitted.
If a variable is a compound of two or more words or abbreviations, each word or abbreviation
should begin with a capital letter.

Temporary variable naming
The name of a temporary variable must be prefixed with the word Temp and not otherwise

TextConst variable names should have a suffix (an approved three-letter suffix: Msg, Tok, Err, Qst,
Lbl, Txt) describing usage.

I guess there is some work to be done :)
